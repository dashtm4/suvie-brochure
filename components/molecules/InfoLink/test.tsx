import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { InfoLink } from '.'

describe('<InfoLink />', () => {
  const href = '/test/'

  it('should renders correct <InfoLink />', () => {
    const wrapper = shallow(<InfoLink href={href} />)

    expect(wrapper.find('Link').prop('href')).toMatch(href)
    expect(wrapper.find('InfoIcon').prop('width')).toEqual(10)
    expect(wrapper.find('InfoIcon').prop('height')).toEqual(10)
    expect(wrapper.find('InfoIcon').prop('colorHex')).toMatch('#000')
    expect(wrapper.find('InfoIcon').prop('className')).toMatch('')

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should InfoIcon has correct width from prop', () => {
    const width = 15

    const wrapper = shallow(<InfoLink href={href} width={width} />)

    expect(wrapper.find('InfoIcon').prop('width')).toEqual(width)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should InfoIcon has correct height from prop', () => {
    const height = 15

    const wrapper = shallow(<InfoLink href={href} height={height} />)

    expect(wrapper.find('InfoIcon').prop('height')).toEqual(height)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should InfoIcon has correct colorHex from prop', () => {
    const colorHex = '#fff'

    const wrapper = shallow(<InfoLink href={href} colorHex={colorHex} />)

    expect(wrapper.find('InfoIcon').prop('colorHex')).toEqual(colorHex)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should InfoIcon has correct className from prop', () => {
    const className = 'test-classname'

    const wrapper = shallow(<InfoLink href={href} className={className} />)

    expect(wrapper.find('InfoIcon').prop('className')).toEqual(className)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
