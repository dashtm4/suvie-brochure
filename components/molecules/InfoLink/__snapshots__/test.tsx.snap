// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoLink /> should InfoIcon has correct className from prop 1`] = `
<Link
  href="/test/"
>
  <a>
    <InfoIcon
      className="test-classname"
      colorHex="#000"
      height={10}
      width={10}
    />
  </a>
</Link>
`;

exports[`<InfoLink /> should InfoIcon has correct colorHex from prop 1`] = `
<Link
  href="/test/"
>
  <a>
    <InfoIcon
      className=""
      colorHex="#fff"
      height={10}
      width={10}
    />
  </a>
</Link>
`;

exports[`<InfoLink /> should InfoIcon has correct height from prop 1`] = `
<Link
  href="/test/"
>
  <a>
    <InfoIcon
      className=""
      colorHex="#000"
      height={15}
      width={10}
    />
  </a>
</Link>
`;

exports[`<InfoLink /> should InfoIcon has correct width from prop 1`] = `
<Link
  href="/test/"
>
  <a>
    <InfoIcon
      className=""
      colorHex="#000"
      height={10}
      width={15}
    />
  </a>
</Link>
`;

exports[`<InfoLink /> should renders correct <InfoLink /> 1`] = `
<Link
  href="/test/"
>
  <a>
    <InfoIcon
      className=""
      colorHex="#000"
      height={10}
      width={10}
    />
  </a>
</Link>
`;
